cli/command/stack/kubernetes/stack.go

```diff
@@ -0,0 +1,106 @@
+package kubernetes
+
+import (
+	"io/ioutil"
+	"path"
+	"sort"
+
+	"github.com/docker/cli/kubernetes/compose/v1beta2"
+	"github.com/docker/cli/kubernetes/labels"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// stack is the main type used by stack commands so they remain independent from kubernetes compose component version.
+type stack struct {
+	name        string
+	composeFile string
+	spec        *v1beta2.StackSpec
+}
+
+// getServices returns all the stack service names, sorted lexicographically
+func (s *stack) getServices() []string {
+	services := make([]string, len(s.spec.Services))
+	for i, service := range s.spec.Services {
+		services[i] = service.Name
+	}
+	sort.Strings(services)
+	return services
+}
+
+// createFileBasedConfigMaps creates a Kubernetes ConfigMap for each Compose global file-based config.
+func (s *stack) createFileBasedConfigMaps(configMaps corev1.ConfigMapInterface) error {
+	for name, config := range s.spec.Configs {
+		if config.File == "" {
+			continue
+		}
+
+		fileName := path.Base(config.File)
+		content, err := ioutil.ReadFile(config.File)
+		if err != nil {
```

Comment:
- Do we want to *group* errors ?
- If a file can't read (but not the first one), a configMap will be created in k8s and not used (orphan) — **and** will probably cause errors next time we do the `docker stack deploy`, right ?

I think we should "read all", and if there is no error, "create all". Then, if one of the creation fails, what should we do with the previous one created ?

Reply:
Right, I'll fix that.

Reply:
I propose to fix this in a follow-up, as we may remove all configmaps and secrets if any error occurs, even while deploying the stack.